benches/suites/raw_node.rs

```diff
@@ -20,3 +26,93 @@ pub fn bench_raw_node_new(c: &mut Criterion) {
 
     c.bench_function("RawNode::new", bench);
 }
+
+pub fn bench_raw_node_leader_propose(c: &mut Criterion) {
+    static KB: usize = 1024;
+    let mut group = c.benchmark_group("RawNode::leader_propose");
+    for size in [
+        0,
+        32,
+        128,
+        512,
+        KB,
+        4 * KB,
+        16 * KB,
+        128 * KB,
+        512 * KB,
+        KB * KB,
+    ]
+    .iter()
+    {
+        group.throughput(Throughput::Bytes(*size as u64));
+        let context = vec![0; 8];
+        let value = vec![0; *size];
+        group.bench_with_input(
+            BenchmarkId::from_parameter(size),
+            &(context, value),
+            |b, (context, value)| {
+                let logger = crate::default_logger();
+                let mut node = quick_raw_node(&logger);
+                node.raft.become_candidate();
+                node.raft.become_leader();
+                b.iter_custom(|iters| {
+                    let mut total = Duration::from_nanos(0);
+                    for _ in 0..iters {
+                        let context = context.to_owned();
+                        let value = value.to_owned();
+                        let start = Instant::now();
+                        node.propose(context, value).expect("");
+                        total = total.add(start.elapsed());
+                    }
+                    total
+                });
+            },
+        );
+    }
+}
+
+pub fn bench_raw_node_new_ready(c: &mut Criterion) {
+    c.bench_function("RawNode::ready", |b: &mut Bencher| {
+        b.iter_custom(|iters| {
+            let logger = crate::default_logger();
+            let mut node = quick_raw_node(&logger);
+            node.raft.become_candidate();
+            node.raft.become_leader();
+            let mut total = Duration::from_nanos(0);
+            for _ in 0..iters {
+                // TODO: Maybe simulate more environments as input. For now, just preparing a raft node after stepping a proposal
+                node.propose(vec![], vec![]).expect("");
+                if node.has_ready() {
+                    let now = Instant::now();
+                    let ready = node.ready();
+                    total = total.add(now.elapsed());
+                    handle_ready(&mut node, ready);
```

Comment:
Handle_ready usually depends on user usage, may not be proper counted in benchmark time.

Reply:
`handle_ready` is excluded from the benchmark as it's called after `total = total.add(now.elapsed())`.

Reply:
Criterion.rs author here! Obviously I don't know much about your code, so feel free to ignore me, but doing `Instant::now()` and `Instant::elapsed()` for every iteration will introduce some nasty measurement overhead and skew your measurements. If you can, setting up a batch of nodes and calling `ready` on all of them using `iter_batched` would give more accurate measurements.

Reply:
Thank you for your guidance! @bheisler
`iter_batched` seems very suitable for the situations I met.

Reply:
Thanks for your amazing library, @bheisler!